### ISSUE TITLE
quantum bibliography style for BibLaTeX

### ISSUE THREAD

**Issue by k4rtik** [cgogolin/quantum-journal#144]
It will be nice to have a `.bbx` file (and others as needed) corresponding to the `.bst` file.

Here's some initial code to add support for the `code` data field as I mentioned at https://github.com/quantum-journal/quantum-journal/pull/143#issuecomment-1279978301:

```diff
diff --git a/quantum.bbx b/quantum.bbx
new file mode 100644
index 0000000..6895974
--- /dev/null
+++ b/quantum.bbx
@@ -0,0 +1,23 @@
+\ProvidesFile{quantum.bbx}[2022/10/16 v0.1 quantum bibliography style]
+\RequireBibliographyStyle{numeric}
+
+\DeclareFieldFormat{code}{\textsc{code:} \url{#1}}
+
+\renewbibmacro*{doi+eprint+url}{%
+  \iftoggle{bbx:doi}
+    {\printfield{doi}}
+    {}%
+  \newunit\newblock
+  \iftoggle{bbx:eprint}
+    {\usebibmacro{eprint}}
+    {}%
+  \newunit\newblock
+  \iftoggle{bbx:url}
+    {\usebibmacro{url+urldate}}
+    {}%
+  \newunit\newblock
+  \iffieldundef{code}
+    {}
+    {\printfield{code}}}
+
+\endinput
diff --git a/quantum.dbx b/quantum.dbx
new file mode 100644
index 0000000..0518606
--- /dev/null
+++ b/quantum.dbx
@@ -0,0 +1,3 @@
+% Teach biblatex about code data field
+\DeclareDatamodelFields[type=field, datatype=uri]{code}
+\DeclareDatamodelEntryfields{code}
```

**Comment by k4rtik** [comment]
_Originally posted by @cgogolin in https://github.com/quantum-journal/quantum-journal/issues/143#issuecomment-1280026668_ :
> The problem with BibLaTeX is that (at leat the last time I checked the arxiv still wasn't able to run BibTeX or Biber and thus a locally generated bbl must be uploaded. Now if that was generated with a BibLaTeX version that doesn't match that of the arXiv things will fail with hard to understand errors.

I think it's fairly easy to solve these days using a tool like [arxiv-collector](https://github.com/djsutherland/arxiv-collector), see particularly:

> Includes the necessary parts of any system package you tell it to upload. By default, this includes biblatex (if you use it) to avoid errors like
>
> > Package biblatex Warning: File '.bbl' is wrong format version



at https://github.com/djsutherland/arxiv-collector/blob/master/README.md#main-features

**Comment by cgogolin** [comment]
Nice work! Can you open a merge request to get started seriously with adding BibLaTeX support? A requirement for me would be that we also add at least some minimal unit tests.

On `arxiv-collector`: This seems like a nice tool for tech savvy users. But there are many authors of Quantum who have never used python or pip and for whom setting up a python environment would be an insurmountable challenge. I am thus very hesitant to recommend or even just suggest such tools to our authors.

This document class and bib style must be as simple to use and as self contained as possible or otherwise we will have many authors running into trouble with their particular variety of the myriad of combinations of TeX distribution, versions of packages and binaries, operating system, and editor/IDE and we will drown in support cases.

**Comment by k4rtik** [comment]
I agree, but I imagine if the user is savvy enough to use `BibLaTeX`, they can figure out the .bbl requirement for arXiv (as I have done before for my submissions to arXiv) even without using the tool.

I will try and create a PR. No timeline yet.
